### PR TITLE
docs: shells for declarative package managers (guix and nix)

### PR DESCRIPTION
**Pull request by pelasgus**
Concerning:
- https://github.com/tauri-apps/wry/pull/1377
- https://github.com/tauri-apps/wry/pull/1373
- https://github.com/tauri-apps/wry/pull/990

> [!NOTE]
> **N.B.**: Im willing to transfer the repo to the organisation for easier management and future updates and maintenance. Its easier overall for it to be a submodule.

> [!IMPORTANT]
> **P.S.**: Will be adding a flake for `nix` as well.

@amrbashir was the original reviewer. please when you have a minute, check this out.

### CODE REVIEW

**Comment by pelasgus** [comment]
If you do take ownership of the repository, then we could perhaps make the shells a little bit more universal so that they could be employed by all tauri repos; thus making development much easier. Otherwise, I can restructure the pr so that instead of a sub-module I add the scripts directly(specialised) to each repo.

File: README.md [hunk 1]
```diff
@@ -83,7 +83,8 @@ Wry also needs [WebKitGTK](https://webkitgtk.org/) for WebView. So please make s
 sudo pacman -S webkit2gtk-4.1
 ```
 
-The `libayatana-indicator` package can be installed from the Arch User Repository (AUR).
+> [!IMPORTANT]
```
Comment (amrbashir) [inline README.md]: let's remove this info, it is outdated and not a concern for this repo anymore

File: README.md [hunk 2]
```diff
@@ -97,7 +98,53 @@ sudo apt install libwebkit2gtk-4.1-dev
 sudo dnf install gtk3-devel webkit2gtk4.1-devel
 ```
 
-Fedora does not have the Ayatana package yet, so you need to use the GTK one, see the [feature flags documentation](https://docs.rs/wry/latest/wry/#feature-flags).
+> [!WARNING]
```
Comment (amrbashir) [inline README.md]: ditto, this info is no longer needed and should be removed

File: README.md [hunk 3]
```diff
@@ -97,7 +98,53 @@ sudo apt install libwebkit2gtk-4.1-dev
...
+
+ ```Nix
+ let
+    # Unstable Channel | Rolling Release
+    pkgs = import (fetchTarball("channel:nixpkgs-unstable")) { };
+    packages = with pkgs; [
+      pkg-config
+      webkitgtk_4_1
+      libayatana-appindicator
+      libappindicator-gtk3
```
Comment (amrbashir) [inline README.md]: these two packages are no longer a concern for this repo, we should remove them from here

File: README.md [hunk 4]
```diff
@@ -97,7 +98,53 @@ sudo apt install libwebkit2gtk-4.1-dev
...
+
+ > [!TIP]
+ > A `manifest.scm` is included in the subdirectory `declerative-shells/etc/shells/guix`. Use as follows:
+ 
+```sh
+ guix shell -m manifest.scm
+ ```
+
+ > [!WARNING]
+ > Guix does not have the Ayatana package yet, so you need to use the GTK one, see the [feature flags documentation](https://docs.rs/wry/latest/wry/#feature-flags).
```
Comment (amrbashir) [inline README.md]: This is not needed and the submodule as well, the instructions at Line 104 are enough

**Review by amrbashir — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -109,6 +140,9 @@ If you are cross-compiling for macOS using [osxcross](https://github.com/tpoecht
 RUSTFLAGS="-l framework=WebKit" cargo build --target=x86_64-apple-darwin --release
 ```
 
+> [!IMPORTANT]
+> If `nix` and/if `nix-darwin` are used then follow instructions from [Nix Section](#nix--nixos).
```
Comment (amrbashir) [inline README.md]: Nix section uses webkit2gtk where on macOS we use the included WKWebView, let's remove this

**Review by amrbashir — APPROVED** [review]

**Pull request merged**